mozci/task.py

```diff
@@ -273,108 +279,25 @@ def __post_init__(self):
             if not is_bad_group(self.id, result.group)
         ]
 
-    def _load_errorsummary(self) -> None:
-        # This may clobber the values that were populated by ActiveData, but
-        # since the artifact is already downloaded, parsed and we need to
-        # iterate over it anyway. It doesn't really hurt and simplifies some
-        # logic. It also ensures we don't attempt to load the errorsummary more
-        # than once.
-        self._results = []
-        self._errors = []
-
-        # Make sure that we don't try to load errorsummary.log for suites which
-        # don't support groups.
-        assert not is_no_groups_suite(self.label)
```

Comment:
I think it'd be good to keep this assertion, maybe move it in the `groups` property?